### DISCUSSION TITLE
How can we acquire a list of courses with maximum enrollment?

### DISCUSSION THREAD

**Discussion by komalm** [Sunbird-Ed/Community#599] [General Discussion]
**Question**: How can we acquire a list of courses with maximum enrollment?
**Details** - We are looking for a list of courses in the following criteria.
1. Live course
2. Open the batch
3. Enrollment maximum

@rajeevsathish, @AmiableAnil

**Comment by Krishnaj20** [comment]
@komalm - I'm not sure I understand your question. What are you looking for? 

@coolbung @amitpriyadarshi

**Reply by komalm** [reply]
@Krishnaj20 - Essentially, we need the maximum enrollment list of courses.

Please let me know if you require any additional explanation on the requirement.

**Reply by Krishnaj20** [reply]
@komalm - sorry I could not understand the question. Are you asking for the maximum number of enrolments possible in a course?

@coolbung @amitpriyadarshi

**Reply by Krishnaj20** [reply]
@komalm - any update on this? 

@coolbung @amitpriyadarshi

**Reply by komalm** [reply]
Hello @Krishnaj20 

I want to get a list of courses sorted by number of enrollments and completions.

**Reply by Krishnaj20** [reply]
@vinod303 - This should be a straightforward query to get this info. Can you help here?

**Reply by vinod303** [reply]
@Krishnaj20 , here the requirement is, suppose there are 10 courses , we need to get them sorted (descending/ascending) order of total enrolments.
As far as I know, Currently we do  not have a single api to achieve this.
We have now, "batch/participants/list"  API and "/api/v1/collection/summary" api , both will take one batch at a time and give count .  So,  if there are more courses , we need to call these  apis as many times to get the count of each of them.

@komalm

**Reply by Krishnaj20** [reply]
@vinod303 - you can configure a new report using hawkeye to capture these details. This is similar to the course consumption report which reports the count of enrolments by batch and location.

**Reply by Krishnaj20** [reply]
@vinod303 @komalm - let us know if there is any additional help needed from the community.

@Abhign @amitpriyadarshi

**Reply by vinod303** [reply]
@Krishnaj20  thanks for suggesting an alternative. I still feel that an API based solution is more easy (as it does not depend on any backed/devops).   
@komalm  pls see  whether you can explore the possibility of a API based solution for that.
